pom.xml

```diff
@@ -308,7 +308,8 @@
                         <configuration>
                             <referenceVersion>5.23ea0</referenceVersion>
                             <artifactsURI>https://teamcity.chronicle.software/repository/download</artifactsURI>
-                            <binaryCompatibilityPercentageRequired>100.0</binaryCompatibilityPercentageRequired>
+                            <!-- Reduce the API surface of ChronicleReader -->
+                            <binaryCompatibilityPercentageRequired>86.1</binaryCompatibilityPercentageRequired>
```

Comment:
This is a significant decrease! @minborg will be happy!

Reply:
It would be interesting to see what kind of API compatibility issues were detected. Could you paste them here in a comment?

Reply:
![Screenshot from 2022-04-01 14-14-14](https://user-images.githubusercontent.com/861395/161188321-a9180370-9a77-45b2-91cd-066f53272ca9.png)

Reply:
I could restore readOne, but I don't know why it was exposed. If it's public it means you can continue to call it beyond when execute() would have stopped calling it. It seems very internal to me, unless I've missed the point of this tool?

Reply:
making applyFiltersAndLog looks like it might have been a way to override what happens to each message, we can do that better by adding a MessageConsumer as a parameter if it's a desired thing?

Reply:
I know customers (and our products) definitely descend from ChronicleReaderMain, but that has not changed.

I believe that customers also descend from ChronicleReader - it is possible they use readOne, but I don't think so. readOne was added in https://github.com/OpenHFT/Chronicle-Queue/commit/8e4d7a5a2d92c8dcc561c25f96ddaccc931b6867 by @peter-lawrey